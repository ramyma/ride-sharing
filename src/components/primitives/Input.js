import styled, { css } from '@xstyled/emotion';
import Box from './Box';

const Input = styled(Box)``;

Input.defaultProps = {
  as: 'input',
  backgroundColor: 'background',
  border: '1px solid',
  borderRadius: 4,
  color: 'inverseBackground',
  height: 'fit-content',
  borderColor: 'inverseBackground',
  p: 2,
  minWidth: 100
};

export default Input;
