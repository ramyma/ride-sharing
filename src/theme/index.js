import { opacify } from 'polished';

const theme = {
  breakpoints: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  },
  defaultColorModeName: 'light',
  colors: {
    background: '#fff',
    grey: '#585858',
    inverseBackground: '#1c1c1c',
    shadow: opacify(0.19, '#000'),
    textDark: '#353535',
    text: '#353535',
    primary: '#ee424a',
    success: '#4CAF50',
    danger: '#f44336',
    warning: '#ffa000',
    modes: {
      dark: {
        background: '#1c1c1c',
        grey: '#9e9e9e',
        inverseBackground: '#fff',
        shadow: opacify(0.19, '#fff'),
        text: '#fff'
      }
    }
  },
  fontSizes: [0, 12, 14, 16, 20, 24, 32, 48, 64, 72],
  sizes: [0, 2, 4, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32],
  shadows: {
    grey: 'rgba(0, 0, 0, 0.19) 0 0 10px 0px',
    greyInset: `inset rgba(0, 0, 0, 0.19) 0 0 10px 0px`,
    pink: '#e42974 0 0 3px',
    text: '0 0 rgba(0, 0, 0, 0.19)'
  },
  space: [0, 4, 8, 16, 24, 48, 96, 144, 192, 240]
};

export default theme;
