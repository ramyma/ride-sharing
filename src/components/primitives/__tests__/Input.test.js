import React from 'react';
import renderer from 'react-test-renderer';
import * as emotion from '@xstyled/emotion';
import { ThemeProvider } from '@xstyled/emotion';
import Input from '../Input';
import theme from '../../../theme';
import { createSerializer } from 'jest-emotion';

expect.addSnapshotSerializer(createSerializer(emotion));

it('Input renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Input value={1} placeholder="test" />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
