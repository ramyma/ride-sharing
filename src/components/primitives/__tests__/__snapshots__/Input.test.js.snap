// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
.emotion-0 {
  background-color: #fff;
  border: 1px solid;
  border-radius: 4px;
  border-color: #1c1c1c;
  color: #1c1c1c;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  min-width: 100px;
  padding: 8px;
}

<input
  className="emotion-0"
  color="inverseBackground"
  height="fit-content"
  placeholder="test"
  value={1}
/>
`;
